src/main/webapp/app/course/learning-goals/learning-goal-management/learning-goal-management.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { LearningGoalService } from 'app/course/learning-goals/learningGoal.service';
+import { JhiAlertService } from 'ng-jhipster';
+import { LearningGoal } from 'app/entities/learningGoal.model';
+import { HttpErrorResponse } from '@angular/common/http';
+import { finalize } from 'rxjs/operators';
+import { onError } from 'app/shared/util/global.utils';
+import { forkJoin, Subject } from 'rxjs';
+import { LearningGoalProgress } from 'app/course/learning-goals/learning-goal-progress-dtos.model';
+
+@Component({
+    selector: 'jhi-learning-goal-management',
+    templateUrl: './learning-goal-management.component.html',
+    styleUrls: ['./learning-goal-management.component.scss'],
+})
+export class LearningGoalManagementComponent implements OnInit, OnDestroy {
+    courseId: number;
+    isLoading = false;
+    learningGoals: LearningGoal[] = [];
+    learningGoalIdToLearningGoalProgress = new Map<number, LearningGoalProgress>();
+    private dialogErrorSource = new Subject<string>();
+    dialogError$ = this.dialogErrorSource.asObservable();
+
+    constructor(private activatedRoute: ActivatedRoute, private router: Router, private learningGoalService: LearningGoalService, private alertService: JhiAlertService) {}
+
+    ngOnDestroy(): void {
+        this.dialogErrorSource.unsubscribe();
+    }
+
+    ngOnInit(): void {
+        this.activatedRoute.params.subscribe((params) => {
+            this.courseId = +params['courseId'];
+            if (this.courseId) {
+                this.loadData();
+            }
+        });
+    }
+
+    identify(index: number, learningGoal: LearningGoal) {
+        return `${index}-${learningGoal.id}`;
+    }
+
+    deleteLearningGoal(learningGoalId: number) {
+        this.learningGoalService.delete(learningGoalId, this.courseId).subscribe(
+            () => {
+                this.dialogErrorSource.next('');
+                this.loadData();
+            },
+            (error: HttpErrorResponse) => this.dialogErrorSource.next(error.message),
+        );
+    }
+
+    getLearningGoalProgress(learningGoal: LearningGoal) {
+        return this.learningGoalIdToLearningGoalProgress.get(learningGoal.id!);
+    }
+
+    loadData() {
+        this.isLoading = true;
+        this.learningGoalService
```

Comment:
This should probably use a resolver instead (which can then be reused for `course-learning-goals` too).

Reply:
No I am strictly against resolvers, they are a bad style imo

Reply:
![image](https://user-images.githubusercontent.com/29718932/101373030-cd032d80-38ac-11eb-8bc0-e32cb202d1bc.png)
https://blog.chinmoymohanty.com/2020/06/23/do-we-really-need-angular-route-resolvers/

Reply:
This was more about fetching the learning goals themselves, but it's not too important here so I won't insist. (Btw, you're setting `isLoading`, which looks like it's blocking to me, too.)